src/main/java/httpserver/HTTPHandler.java

```diff
@@ -22,232 +22,187 @@
+     * catchall route, a 501 (Not implemented) is sent to the client.
+     *
+     * @param request     The incoming HTTPRequest.
+     * @param response    The outgoing HTTPResponse, waiting to be filled by an
+     *                    HTTPHandler.
+     *
+     * @see HTTPHandler#addGET
+     * @see HTTPHandler#addPOST
+     * @see HTTPHandler#addDELETE
+     * @see HTTPResponse#NOT_A_METHOD_ERROR
+     */
+    public void handle(HTTPRequest request, HTTPResponse response) {
+	String httpRequestType = request.getRequestType().toUpperCase();
+	if (!routes.containsKey(httpRequestType)) {
+	    response.message(404, "No " + httpRequestType + " routes exist.");
+	    return;
+	}
+
+	MethodWrapper finalMethod = null;
+	int bestFit = 0;
+	for (MethodWrapper method : routes.get(httpRequestType)) {
+	    if (method.matchesPerfectly(request.getSplitPath())) {
+			finalMethod = method;
+			break;
+	    }
+
+	    int testScore = method.howCorrect(request.getSplitPath());
+	    if (testScore > bestFit) {
+			finalMethod = method;
+			bestFit = testScore;
+			}
+		}
+
+		if (finalMethod == null) {
+			response.message(501, HTTPResponse.NOT_A_METHOD_ERROR);
+			return;
+		}
+
+		finalMethod.invoke(request, response, request.getSplitPath());
+    }
+
+    /**
+     * Attach a method to a GET request at a path. <p>
+     *
+     * Methods are passed in as a String, and must be a member of the current
+     * handler.<p>
+     *
+     * Path's should come in "/path/to/action" form. If the method requires
+     * any parameters that aren't an HTTPResponse, HTTPRequest, or Map,
+     * they should be included in the path, in the order they're
+     * listed in the method header, in "{ClassName}" form. Example:
+     * <code>/hello/{String}/{String}</code> is a good path. <p>
+     *
+     * Methods being passed in must accept an HTTPResponse as their first
+     * parameter. Methods may optionally accept an HTTPRequest and a
+     * Map&lt;String, String&gt; in that order (they may accept a Map but not an
+     * HTTPRequest, but if they accept both the HTTPRequest must come first).
+     *
+     * Parameters following the above must be included in the java.lang library
+     * and have a constructor that takes in a String.
+     * Any other type of parameter will cause an exception to occur. <p>
+     *
+     * Additionally, primitives are not permited, because they're not classes in
+     * the java.lang library. The three most common parameter types are String,
+     * Integer, and Double.
+     *
+     * @param path        Path to match
+     * @param methodName  Method belonging to the current class, in String form.
+     * @throws HTTPException When you do bad things.
+     *
+     * @see HTTPHandler#addPOST
+     * @see HTTPHandler#addDELETE
+     * @see HTTPResponse
+     * @see HTTPRequest
+     */
+    public void get(String path, Route route) throws HTTPException {
+		addRoute(HTTPRequest.GET_REQUEST_TYPE, path, route);
```

Comment:
Side comment: should there be a `HTTPMethods` class, which is just a collection of constants.

Basically you'd say `HTTPMethods.GET` instead of `HTTPRequest.GET_REQUEST_TYPE`.